### PR TITLE
Issue 363 transform api

### PR DESCRIPTION
**Pull request by kain88-de**
See #1010, #1022, #1025 

Changes made in this Pull Request:
 - added tests
 - clean up docstrings

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/groups.py [hunk 1]
```diff
@@ -509,29 +511,30 @@ def rotate(self, R):
...
-        atomgroup = self.atoms
 
-        R = np.matrix(R, copy=False, dtype=np.float32)
+        """
+        R = np.asarray(R)
         # changes the coordinates (in place)
-        x = atomgroup.universe.trajectory.ts.positions
-        idx = atomgroup.indices
-        x[idx] = x[idx] * R.T  # R.T acts to the left & is broadcasted N times.
+        x = self.atoms.unique.universe.trajectory.ts.positions
```
Comment (kain88-de) [inline package/MDAnalysis/core/groups.py]: all the other functions that I've tested here also use `unique` so I just followed their style.

**Comment by dotsdl** [comment]
Related: I tried to deprecate this behavior in the last release, but there was some argument against this. I'm partial to your view, though, that robust simplicity is better than convenient but confusing complexity. See the discussion on #698.

**Comment by kain88-de** [comment]
I'm done now. transform/translate/rotate/rotateby are now fully tested as well.

**Comment by richardjgowers** [comment]
cool thanks!

**Pull request merged**